src/ripple/app/ledger/LedgerToJson.h

```diff
@@ -129,49 +132,84 @@ void fillJson (Object& json, LedgerFill const& fill)
             {
                 if (type == SHAMapTreeNode::tnTRANSACTION_NM)
                 {
-                    SerializerIterator sit (item->peekSerializer ());
-                    STTx txn (sit);
-                    txns.append (txn.getJson (0));
+                     if (bBinary)
+                    {
+                        auto&& obj = appendObject (txns);
+                        obj[jss::tx_blob] = strHex (item->peekData ());
+                    }
```

Comment:
@rec Hold on, I think I was confused (I was looking at the wrong AppendObject function - one that returned on Object - not a Json::Value&).

On the other hand, if it's returning a ref, why use a universal reference? And if it sometimes returns a value the original objection stands.